### PR TITLE
When setting JS values, do so in a batch.

### PR DESCRIPTION
**Pull request by brian-brazil**
This makes loading saves much faster, and
avoids some issues like flickering output
as things settles.

Also fixed some formatting in passing.

### CODE REVIEW

**Comment by frank-weinberg** [comment]
This should also be done around Start New Game.

And unless I'm missing something, if we also wrap `ScoreBoardClock.updateClients()`, we could remove all other calls to `reqestBatch*`

**Comment by brian-brazil** [comment]
I've adjusted that. Clocks are already covered, but Media needed adding.

File: src/com/carolinarollergirls/scoreboard/event/ScoreBoardEventProviderImpl.java [hunk 1]
```diff
@@ -112,6 +113,15 @@ protected void requestBatchEnd() {
         scoreBoardChange(new ScoreBoardEvent(this, BatchEvent.END, Boolean.TRUE, Boolean.TRUE));
     }
 
+    public void runInBatch(Runnable r) {
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/event/ScoreBoardEventProviderImpl.java:117]: This should be tagged `@Override`

File: src/com/carolinarollergirls/scoreboard/jetty/LoadJsonScoreBoard.java [hunk 2]
```diff
@@ -64,12 +62,20 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
         }
     }
 
-    protected void handleJSON(HttpServletRequest request, HttpServletResponse response, JSONObject json) throws IOException {
+    protected void handleJSON(HttpServletRequest request, HttpServletResponse response, final JSONObject json) throws IOException {
         if (request.getPathInfo().equalsIgnoreCase("/load")) {
-            scoreBoard.reset();
-            ScoreBoardJSONSetter.set(scoreBoard, json);
+            scoreBoard.runInBatch(new Runnable() {
+                public void run() {
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/jetty/LoadJsonScoreBoard.java:69]: This should be tagged `@Override`

File: src/com/carolinarollergirls/scoreboard/jetty/LoadJsonScoreBoard.java [hunk 3]
```diff
@@ -64,12 +62,20 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
...
+                    scoreBoard.reset();
+                    ScoreBoardJSONSetter.set(scoreBoard, json);
+                }
+            });
         } else if (request.getPathInfo().equalsIgnoreCase("/merge")) {
-            ScoreBoardJSONSetter.set(scoreBoard, json);
+            scoreBoard.runInBatch(new Runnable() {
+                public void run() {
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/jetty/LoadJsonScoreBoard.java:77]: This should be tagged `@Override`

File: src/com/carolinarollergirls/scoreboard/jetty/WS.java [hunk 4]
```diff
@@ -123,26 +123,34 @@ public synchronized void onMessage(String message_data) {
                     String f = json.getString("flag");
                     if ("reset".equals(f)) { flag = Flag.RESET; }
                     if ("change".equals(f)) { flag = Flag.CHANGE; }
-                    ScoreBoardJSONSetter.JSONSet js = new ScoreBoardJSONSetter.JSONSet(key, v, flag);
-                    ScoreBoardJSONSetter.set(sb, Collections.singletonList(js));
+                    final ScoreBoardJSONSetter.JSONSet js = new ScoreBoardJSONSetter.JSONSet(key, v, flag);
+                    sb.runInBatch(new Runnable() {
+                        public void run() {
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/jetty/WS.java:129]: This should be tagged `@Override`

File: src/com/carolinarollergirls/scoreboard/jetty/WS.java [hunk 5]
```diff
@@ -123,26 +123,34 @@ public synchronized void onMessage(String message_data) {
...
-                    PreparedTeam t1 = sb.getPreparedTeam(data.getString("Team1"));
-                    PreparedTeam t2 = sb.getPreparedTeam(data.getString("Team2"));
-                    String rs = data.getString("Ruleset");
-                    sb.reset();
-                    sb.getRulesets().setCurrentRuleset(rs);
-                    sb.getTeam(Team.ID_1).loadPreparedTeam(t1);
-                    sb.getTeam(Team.ID_2).loadPreparedTeam(t2);
+                    final JSONObject data = json.getJSONObject("data");
+                    sb.runInBatch(new Runnable() {
+                        public void run() {
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/jetty/WS.java:137]: This should be tagged `@Override`

File: src/com/carolinarollergirls/scoreboard/core/impl/BoxTripImpl.java [hunk 6]
```diff
@@ -147,12 +147,9 @@ protected void itemRemoved(AddRemoveProperty prop, ValueWithId item) {
     @Override
     public void execute(CommandProperty prop) {
         synchronized (coreLock) {
-            requestBatchStart();
             switch((Command)prop) {
             case DELETE:
                 unlink();
-                ((ScoreBoardEventProviderImpl) parent).scoreBoardChange(
-                        new ScoreBoardEvent(this, BatchEvent.END, Boolean.TRUE, Boolean.TRUE));
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/core/impl/BoxTripImpl.java:158]: `ScoreBoardEvent` is now unused

File: src/com/carolinarollergirls/scoreboard/core/impl/JamImpl.java [hunk 7]
```diff
@@ -81,17 +81,12 @@ public void execute(CommandProperty prop) {
                         (this == ((Period)parent).getCurrentJam())) {
                     break;
                 }
-                requestBatchStart();
                 unlink();
                 scoreBoard.updateTeamJams();
-                ((ScoreBoardEventProviderImpl) parent).scoreBoardChange(
-                        new ScoreBoardEvent(this, BatchEvent.END, Boolean.TRUE, Boolean.TRUE));
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/core/impl/JamImpl.java:88]: `ScoreBoardEvent` and `ScoreBoardEventProviderImpl` are now unused

**Comment by brian-brazil** [comment]
All fixed.

**Pull request merged**